src/Application/Handlers/HttpErrorHandler.php

```diff
@@ -62,12 +61,31 @@ protected function respond(): Response
         }
 
         $payload = new ActionPayload($statusCode, null, $error);
-        $encodedPayload = json_encode($payload, JSON_PRETTY_PRINT | JSON_THROW_ON_ERROR);
+        try {
+            $encodedPayload = json_encode($payload, JSON_PRETTY_PRINT | JSON_THROW_ON_ERROR);
+        } catch (\JsonException $_exception) {
+            $this->logger->warning('Original error is not JSON so cannot be returned verbatim');
+            $payload = new ActionPayload(
+                $statusCode,
+                null,
+                $this->createInternalError('Original error is not encodable as JSON, see info logs'),
```

Comment:
Sorry for doing two suggestions separately, but I think we'd want the JSON exception message too. Maybe this?

```suggestion
        } catch (\JsonException $exception) {
            $this->logger->warning('Original error is not JSON so cannot be returned verbatim');
            $payload = new ActionPayload(
                $statusCode,
                null,
                $this->createInternalError('JsonException while handling error: {$exception->getMessage()}'),
```

Not sure why we need the log warning as well - doesn't returning the SERVER_ERROR type response automatically make an error log? If not seems like this should be an error rather than a warning anyway.

Reply:
I can't see anything in the shutdown handler that does that, but I might have forgotten about something that does.

I'm not sure an error is the ultimate outcome we want though – if some types of invalid input can lead our libraries to surface unencode-able messages, and we know this is inevitable at low levels, then normally we'd want to only log a warning at most. This gets a bit messy because I'm not sure how we can tell if it was definitely the same source of a `\JsonException` that we've seen before.

Reply:
In principle we should almost certainly reject that input, but maybe it's not worth the effort of trying to detect it as long as we're relying on the built in PHP `string`. But we would want to log the message from the exception as part of the warning so we can decide, wouldn't we?

Reply:
Yep good point. I think where it was added in the suggestion would cause a 2nd JSON exception unless I'm missing something. Added to the warning log in https://github.com/thebiggive/identity/pull/167/commits/15ea916ce2e884eabca1c6cae52b70099b88fa8c